# Dependency removal cleanup patch
#
# Copyright (C) 2006 Nokia Corporation.
# This file is distributed under the same license as the gtkhtml package
#
diff -Nru gtkhtml-3.9.1/Makefile.am gtkhtml-3.9.1-fixed/Makefile.am
--- gtkhtml-3.9.1/Makefile.am	2006-01-02 12:04:05.000000000 +0200
+++ gtkhtml-3.9.1-fixed/Makefile.am	2006-01-02 13:26:10.000000000 +0200
@@ -44,3 +44,17 @@
 	intltool-update.in			\
 	intltool-merge.in
 
+#AK/22-12-05: common way to create .deb packages (Scratchbox)
+#
+deb:
+	dpkg-buildpackage -rfakeroot -uc -us -sa -D
+
+# Use "dch -a" to add comment to debian/changelog
+# Use "dch -v xx" to add new version to debian/changelog
+#
+# DEBEMAIL and DEBFULLNAME better be defined...! :)
+#
+# DO NOT MANUALLY EDIT THE debian/changelog.
+#
+ver:
+	dch -v 999
diff -Nru gtkhtml-3.9.1/autogen.sh gtkhtml-3.9.1-fixed/autogen.sh
--- gtkhtml-3.9.1/autogen.sh	1970-01-01 02:00:00.000000000 +0200
+++ gtkhtml-3.9.1-fixed/autogen.sh	2006-01-02 13:26:09.000000000 +0200
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+set -x
+aclocal
+autoconf
+autoheader
+automake --add-missing --foreign
diff -Nru gtkhtml-3.9.1/configure.in gtkhtml-3.9.1-fixed/configure.in
--- gtkhtml-3.9.1/configure.in	2006-01-02 12:04:05.000000000 +0200
+++ gtkhtml-3.9.1-fixed/configure.in	2006-01-02 13:26:09.000000000 +0200
@@ -74,8 +74,8 @@
 dnl
 GAIL_MODULES=""
 AC_ARG_ENABLE(a11y,
-	      [ --enable-a11y	Enable Accessibility (requires gail) [default=yes]],
-	      [ac_cv_enable_a11y=$enableval],[ac_cv_enable_a11y=yes])
+	      [ --enable-a11y	Enable Accessibility (requires gail) [default=no]],
+	      [ac_cv_enable_a11y=$enableval],[ac_cv_enable_a11y=no])
 AC_MSG_CHECKING([whether to enable a11y])
 if test "$ac_cv_enable_a11y" = yes; then
 	AC_MSG_RESULT(yes)
@@ -101,8 +101,8 @@
 dnl
 GNOMEPRINT_MODULES=""
 AC_ARG_ENABLE(printing,
-	      [ --enable-printing Enable Printing (requires gnome-print) [default=yes]],
-	      [ac_cv_enable_printing=$enableval],[ac_cv_enable_printing=yes])
+	      [ --enable-printing Enable Printing (requires gnome-print) [default=no]],
+	      [ac_cv_enable_printing=$enableval],[ac_cv_enable_printing=no])
 AC_MSG_CHECKING([whether to enable printing])
 if test "$ac_cv_enable_printing" = yes; then
 	AC_MSG_RESULT(yes)
@@ -122,8 +122,8 @@
 dnl
 GNOME_MODULES=""
 AC_ARG_ENABLE(gnome,
-	      [ --enable-gnome Enable Gnome  (requires many things) [default=yes]],
-	      [ac_cv_enable_gnome=$enableval],[ac_cv_enable_gnome=yes])
+	      [ --enable-gnome Enable Gnome  (requires many things) [default=no]],
+	      [ac_cv_enable_gnome=$enableval],[ac_cv_enable_gnome=no])
 AC_MSG_CHECKING([whether to enable gnome])
 if test "$ac_cv_enable_gnome" = yes; then
 	AC_MSG_RESULT(yes)
@@ -161,8 +161,8 @@
 dnl 
 EDITOR_MODULES=""
 AC_ARG_ENABLE(editor,
-	      [ --enable-editor Enable Editor  (requires many things) [default=yes]],
-	      [ac_cv_enable_editor=$enableval],[ac_cv_enable_editor=yes])
+	      [ --enable-editor Enable Editor  (requires many things) [default=no]],
+	      [ac_cv_enable_editor=$enableval],[ac_cv_enable_editor=no])
 AC_MSG_CHECKING([whether to enable editor])
 if test "$ac_cv_enable_editor" = yes; then
 	AC_MSG_RESULT(yes)
@@ -191,16 +191,26 @@
 dnl **********************************
 dnl soup for testgtkhtml
 dnl **********************************
-
-PKG_CHECK_MODULES(SOUP, libsoup-2.2 >= 2.1.6, have_soup="yes", have_soup="no")
-if test "x$have_soup" = "xno"; then
-	PKG_CHECK_MODULES(SOUP, soup-2.0 >= 0.7.9, [have_soup="yes"
+AC_ARG_ENABLE(soup,
+	      [ --enable-soup Enable soup [default=no]],
+	      [ac_cv_enable_soup=$enableval],[ac_cv_enable_soup=no])
+AC_MSG_CHECKING([whether to enable soup])
+if test "$ac_cv_enable_soup" = yes; then
+	AC_MSG_RESULT(yes)
+	PKG_CHECK_MODULES(SOUP, libsoup-2.2 >= 2.1.6, have_soup="yes", have_soup="no")
+	if test "x$have_soup" = "xno"; then
+		PKG_CHECK_MODULES(SOUP, soup-2.0 >= 0.7.9, [have_soup="yes"
 						    AC_DEFINE(HAVE_OLD_SOUP)],
 						   have_soup="no")
+	fi
+	AC_SUBST(SOUP_CFLAGS)
+	AC_SUBST(SOUP_LIBS)
+else
+	have_soup="no"
+	AC_MSG_RESULT(no)
+	AC_DEFINE(DISABLE_SOUP)
 fi
 AM_CONDITIONAL(HAVE_SOUP, test x$have_soup != xno)
-AC_SUBST(SOUP_CFLAGS)
-AC_SUBST(SOUP_LIBS)
 
 dnl **************************************************
 dnl * File Chooser
@@ -283,7 +293,8 @@
 	accessibility support:		$ac_cv_enable_a11y
 	gnome-print support:            $ac_cv_enable_printing
 	gnome support:			$ac_cv_enable_gnome
-	editor component:               $ac_cv_enable_editor	
+	editor component:               $ac_cv_enable_editor
+	soup:				$ac_cv_enable_soup	
 		
 	Editor component type:		${GNOME_GTKHTML_EDITOR_TYPE}
 "
